src/components/expression/expression-renderer.js

```diff
@@ -673,23 +895,68 @@ genotet.ExpressionRenderer.prototype.highlightLabelsForClickedCell_ = function(
 };
 
 /**
+ * Unhighlights the label of the clicked cells for the heatmap.
+ * @param {!genotet.ExpressionRenderer.Cell} cell
+ * @private
+ */
+genotet.ExpressionRenderer.prototype.highlightLabelsForClickedCell_ =
+  function(cell) {
+    this.svgGeneLabels_
+      .selectAll('text')
+      .classed('click-highlighted', function(d, i) {
+        return cell.row == i;
+      });
+    this.svgConditionLabels_
+      .selectAll('text')
+      .classed('click-highlighted', function(d, i) {
+        return cell.column == i;
+      });
+  };
+
+/**
+ * Unhighlights all the labels of cells for the heatmap.
+ * @private
+ */
+genotet.ExpressionRenderer.prototype.unhighlightLabelsForClickedCell_ =
+  function() {
+    this.svgGeneLabels_
+      .selectAll('text')
+      .classed('click-highlighted', false);
+    this.svgConditionLabels_
+      .selectAll('text')
+      .classed('click-highlighted', false);
+  };
+
+/**
  * Highlights the labels of clicked profile.
- * @param {*} profile TODO(liana): fill the type.
+ * @param {!genotet.ExpressionRenderer.Profile} profile
  * @private
  */
 genotet.ExpressionRenderer.prototype.highlightLabelsForClickedProfile_ =
-    function(profile) {
-  this.svgGeneLabels_.selectAll('text')
-    .attr('fill', 'black');
-  this.svgConditionLabels_.selectAll('text')
-    .attr('fill', 'black');
-  this.svgGeneLabels_.select('text:nth-child(' +
-      (profile.row + 1) + ')')
-    .attr('fill', profile.color);
-  this.svgConditionLabels_.select('text:nth-child(' +
-      (profile.hoverColumn + 1) + ')')
-    .attr('fill', profile.color);
-};
+  function(profile) {
+    this.unhighlightLabelsForClickedProfile_();
+    this.svgGeneLabels_.select('text:nth-child(' + (profile.row + 1) + ')')
+      .attr('fill', profile.color)
```

Comment:
According to W3C recommendations fill is a _property_. We consistently use style to set _properties_  to distinguish them from non-property attributes such as "r", "cx", etc.

Your way of using both attr and style doesn't make much sense to me as you are giving a same element two rendering styles at the same time. The two styles are competing. Competing styles should be avoided as much as possible, e.g. in CSS you want to avoid [Specifity Wars](https://stuffandnonsense.co.uk/archives/css_specificity_wars.html).

If you want to highlight interacted elements you can simply add classes or use ":state" selector.